src/language-usability.rst

```diff
@@ -16,6 +16,171 @@ Usability of Type Systems
+User Studies
+------------
+
+..
+  Outline
+  * User study studying global usability of APIs very difficult, so many factors
+  * Most user studies in API examine a specific aspect of api design and test it
+  * Paper: Usability Implications of Requiring Parameters in Objects' Constructors
+  * Paper: The Implications of Method Placement on API Learnability
+..
+
+Testing how to best design APIs is immensely difficult, as the numerous API design
+decisions all affect the overall usability interdependently
+:cite:`myers2016improving`. Testing the high-level and global design decisions
+is often intractable because of this. Most user studies in API usability
+examine a specific aspect of API design :cite:`stylos2007usability`
+:cite:`stylos2008implications`.
+
+.. container:: bib-item
+
+  .. bibliography:: api-usability.bib
+    :filter: key == 'stylos2007usability'
+
+  This paper examines the impact of requiring parameters in
+  objects' constructors on usability. The study in this paper showed that developers
+  strongly preferred and were more effective with APIs that did not
+  require parameters in object constructors. This contradicted the
+  belief that object constructor parameters would "self-document"
+  themselves and guide developers to using the objects correctly.
+
+.. container:: bib-item
+
+  .. bibliography:: api-usability.bib
+    :filter: key == 'stylos2008implications'
+
+  This paper examines the impact of method placement on usability.
+  In the user study participants were given nearly identical APIs but
+  the comparison group contained a method that was placed in an
+  unexpected location. Participants with "correctly-placed" methods
+  were orders of magnitudes faster in their tasks.
+
+Documentation Improving Tools
+-----------------------------
+
+..
+  Outline
+  * API usability can also be improved by creating better tools for understanding and searching API's -- documentation
+  * Another distinction with language design - documentation is part of API. Using API is using documentation
+  * Paper: Jadeite
+  * Paper: Improving API Documentation Usability with Knowledge Pushing
+  * Paper: Mica: A Web-Search Tool for Finding API Components and Examples
+..
+
+API usability can also be improved by creating better tools for understanding and searching through APIs -- improving API documentation
+:cite:`stylos2009improving` :cite:`dekel2009improving` :cite:`stylos2006mica`.
+Documentation represents another distinction of API design from language design.
+The usage of APIs is so closely tied with using the API documentation that
+the documentation can be considered part of the API.
+A good documentation is integral in making an inherently complex API more learnable.
+
+.. container:: bib-item
+
+  .. bibliography:: api-usability.bib
+    :filter: key == 'stylos2009improving'
+
+  This paper presents Jadeite, a tool for improving API documentation by
+  letting API users add "pretend" classes or methods in places where
```

Comment:
It isn't totally clear what this means, can you elaborate a bit in this description?